### ISSUE TITLE
Getting "Cannot create buckets using a POST" when query storage_polices with GCP storage disk

### ISSUE THREAD

**Issue by boqu** [ClickHouse/ClickHouse#24246]
**Describe the bug**
We are trying to run Clickhouse in GCP with GCP storage as one of disk in storage policy. See below storage policy config. When query system.storage_polices, I'm getting "Cannot create buckets using a POST" error. Meanwhile there are a lot of objects, which contain "test" string created in the bucket.  I didn't find anything particularly in document regarding using GCP storage with Clickhouse. 

**How to reproduce**
* v21.3.8
* clickhouse-client console.
* Storage policy
```
<storage_configuration>
         <disks>
         <data1>
            <path>/mnt/data1/clickhouse/data/</path>
         </data1>
         <data2>
            <path>/mnt/data2/clickhouse/data/</path>
         </data2>
         <bucket1>
           <type>s3</type>
             <endpoint>https://storage.googleapis.com/bucket_name/root_path/</endpoint>
             <access_key_id>######</access_key_id>
             <secret_access_key>######</secret_access_key>
         </bucket1>
       </disks>
       <policies>
         <logs_policy>
           <volumes>
             <data>
               <disk>data1</disk>
               <disk>data2</disk>
             </data>
             <s3>
               <disk>bucket1</disk>
             </s3>
           </volumes>
         </logs_policy>
       </policies>
   </storage_configuration>
```

* Run `select * from system.storage_policies;` statemewnt

**Expected behavior**
It should print out all storage polices details.

**Error message and/or stacktrace**
```
SELECT *
FROM system.storage_policies

Query id: 3885079a-e70b-439c-91df-a7861840e241

Cannot load data for command line suggestions: Code: 499, e.displayText() = DB::Exception: Received from localhost:9000. DB::Exception: 99: <?xml version='1.0' encoding='UTF-8'?><Error><Code>InvalidArgument</Code><Message>Invalid argument.</Message><Details>Cannot create buckets using a POST.</Details></Error>. (version 21.3.8.1)

0 rows in set. Elapsed: 63.829 sec.

Received exception from server (version 21.3.8):
Code: 499. DB::Exception: Received from localhost:9000. DB::Exception: 99: <?xml version='1.0' encoding='UTF-8'?><Error><Code>InvalidArgument</Code><Message>Invalid argument.</Message><Details>Cannot create buckets using a POST.</Details></Error>.
```

Errors in the server log:
```
2021.05.18 10:38:09.748502 [ 1784 ] {} <Error> AWSClient: HTTP response code: 400
Resolved remote host IP address:
Request ID:
Exception name:
Error message: <?xml version='1.0' encoding='UTF-8'?><Error><Code>InvalidArgument</Code><Message>Invalid argument.</Message><Details>Cannot create buckets using a POST.</Details></Error>
7 response headers:
connection : close
content-length : 171
content-type : application/xml; charset=UTF-8
date : Tue, 18 May 2021 10:38:09 GMT
server : UploadServer
vary : Origin
```

**Additional context**
N/A

**Comment by joschi** [comment]
It looks like GCS doesn't support the [DeleteObjects](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html ) request and possibly other batch APIs which is being used by ClickHouse:

[Google Cloud Storage Issue Tracker: Support deleteObjects API](https://issuetracker.google.com/issues/162653700)

https://github.com/ClickHouse/ClickHouse/blob/30c04f1fb93728f9812c397440f0d66545cfe882/src/Disks/S3/DiskS3.cpp#L49-L59

Same with [ListObjectsV2](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html):
> Note: While most actions are interoperable with the Amazon S3 V2 SDK, listing objects can only be performed using the Amazon S3 V1 list objects method.

Source: https://cloud.google.com/storage/docs/interoperability

https://github.com/ClickHouse/ClickHouse/blob/30c04f1fb93728f9812c397440f0d66545cfe882/src/Disks/S3/DiskS3.cpp#L26

**Comment by kamalmarhubi** [comment]
ListObjectsV2 was added as a public preview in July: https://cloud.google.com/storage/docs/release-notes#July_12_2021

Issue tracking it: https://issuetracker.google.com/issues/155109631

Are there other methods missing in GCP beyond that and the already mentioned DeleteObjects method?

**Comment by MyroTk** [comment]
I was unable to reproduce this issue on 22.3.7. 

With the same storage config, the select query executes as expected.
```
clickhouse1 :) select * from system.storage_policies

SELECT *
FROM system.storage_policies

Query id: 1b869688-f340-42a4-abd6-818b9f687511

┌─policy_name─┬─volume_name─┬─volume_priority─┬─disks─────────────┬─volume_type─┬─max_data_part_size─┬─move_factor─┬─prefer_not_to_merge─┐
│ default     │ default     │               1 │ ['default']       │ JBOD        │                  0 │           0 │                   0 │
│ logs_policy │ data        │               1 │ ['data1','data2'] │ JBOD        │                  0 │         0.1 │                   0 │
│ logs_policy │ s3          │               2 │ ['bucket1']       │ JBOD        │                  0 │         0.1 │                   0 │
└─────────────┴─────────────┴─────────────────┴───────────────────┴─────────────┴────────────────────┴─────────────┴─────────────────────┘

3 rows in set. Elapsed: 0.004 sec.
```

**Comment by azat** [comment]
I don't see that `DeleteObjects` is supported. @filimonov I think this had been closed too early.

**Comment by azat** [comment]
>I don't see that DeleteObjects is supported

Ok, it requires `support_batch_delete=false` setting (added in #37659)

**Comment by azat** [comment]
Also, it is interesting that before merging #37139, this error was only appears in logs, and does not the request.

**Comment by alexey-milovidov** [comment]
I don't like if it requires additional configuration rather than being supported automatically.

**Comment by azat** [comment]
> I don't like if it requires additional configuration rather than being supported automatically.

It will be resolved by #37882.

Also since this is the issue that has some important patches for GCS support, right now you cannot use http proxy with https GCS URI, it requires https://github.com/ClickHouse/poco/pull/63 (and subsequent submodule update).

**Comment by azat** [comment]
#37882 had been merged (and also #38726), this can be closed.
